products/editions/enterprise-edition/b2b-suite/guides/storefront/ajax-panel.md

```diff
@@ -0,0 +1,134 @@
+# Ajax panel
+
+You can download a plugin showcasing the topic [here](/products/editions/enterprise-edition/b2b-suite/example-plugins/B2bAjaxPanel.zip).
+
+## Table of contents
+
+* [Description](#description)
+* [Basic usage](#basic-usage)
+* [Extended usage](#extended-usage)
+  * [Make links clickable](#make-links-clickable)
+  * [Ignore links](#ignore-links)
+  * [Link to a different panel](#link-to-a-different-panel)
+* [Ajax Panel Plugins](#ajax-panel-plugins)
+  * [Modal](#modal)
+  * [TriggerReload](#triggerreload)
+  * [TreeSelect](#treeselect)
+
+## Description
+
+`AjaxPanel` is a jQuery based extension to the Shopware Frontend Framework. It mimics `iFrame` behaviour by integrating content from different controller actions through ajax into a single view and intercepting, usually page changing, events and transforming them into XHR-Requests.
+
+The diagram below shows how this schematically behaves:
+
+![image](/.gitbook/assets/ajax-panel-abstract.svg)
+
+## Basic usage
+
+The `AjaxPanel` plugin is part of the b2b frontend and will scan your page automatically for the trigger class `b2b--ajax-panel`. The most basic ajax panel looks like this:
+
+```html
+<div
+    class="b2b--ajax-panel"
+    data-url="{url controller="b2bcontact" action="grid"}"
+>
+    <!-- will load content here -->
+</div>
+```
+
+After `$(document).ready()` is triggered it will trigger a XHR GET-Request and replace it's inner html with the responses content. Now all clicks on links and form submits inside the container will be changed to XHR-Requests. A streamlined example of this behaviour can be found in the [B2BAjaxPanel Example Plugin](/products/editions/enterprise-edition/b2b-suite/example-plugins/B2bAjaxPanel.zip), but it is used across the B2B-Suite.
```

Comment:
We are not using jQuery anymore, but the idea is still correct. You could just replace this with "After the document is ready, the ajax panel will trigger ...."